lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -450,4 +464,192 @@ def _env_manager_for(self, type: ServiceType) -> _EnvManager:
 
     def munge_key_manager(self) -> MungeKeyManager:
         """Get the `MungekeyManager` class of this ops manager."""
-        return _SnapMungeKeyManager()
+        return _MungectlManager("slurm.mungectl")
+
+
+# ========================= deb manager =========================
+
+
+class _SystemctlServiceManager(ServiceManager):
+    """Control a Slurm service using systemctl services."""
+
+    def __init__(self, service: ServiceType) -> None:
+        self._service = service
+
+    def enable(self) -> None:
+        """Enable service.
+
+        Raises:
+            SlurmOpsError: Raised if `systemctl enable ...` returns a non-zero returncode.
+        """
+        _call("systemctl", "enable", "--now", self._service.value)
+
+    def disable(self) -> None:
+        """Disable service."""
+        _call("systemctl", "disable", "--now", self._service.value)
+
+    def restart(self) -> None:
+        """Restart service."""
+        _call("systemctl", "reload-or-restart", self._service.value)
+
+    def active(self) -> bool:
+        """Return True if the service is active."""
+        cmd = ["systemctl", "is-active", "--quiet", self._service.value]
+        _logger.debug(f"Executing command {cmd}")
+        proc = subprocess.run(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True,
+            bufsize=1,
+            encoding="utf-8",
+        )
+        _logger.debug(f"command {cmd} exit code: {proc.returncode}. output:\n{proc.stdout}")
+        return proc.returncode == 0
```

Comment:
[nit]: I'm wondering if we should potentially just modify `_call` here to also return the exit code of a subprocess as well? Like `Tuple[str, int]`. That way we don't need a different way for calling `is-active`. Let me know your thoughts here.

_I know that this method is based off of the implementation from the systemd charm library_ :sweat_smile: